### DISCUSSION TITLE
oo/encapsulation-inheritance-polymorphism

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#205] [Announcements]
# oo/encapsulation-inheritance-polymorphism

本文详细解析Java面向对象编程的三大核心特性：封装、继承和多态。文章将分别介绍这三个特性的概念、原理以及在实际开发中的应用场景，通过实例帮助读者更好地理解和掌握Java面向对象编程的基本技巧。

https://tobebetterjavaer.com/oo/encapsulation-inheritance-polymorphism.html

**Comment by thehuanxie** [comment]
拉面馆小人儿不是uml man吗hhhhhhhhhhhh

**Comment by Eivae** [comment]
Wangxiaosan那段代码，父类构造方法中的write()为什么调用的是子类的write方法而不是父类中的write方法？不是父类先创建吗？

**Reply by ThePHOENlX** [reply]
因为new的是Wangxiaosan类，他重写了父类的write（）方法，所以调用的是子类的write（）。不然重写就没意义了
